### PR TITLE
Add new tools: ekstctl

### PR DESCRIPTION
**Pull request by Yannig**
This tools help to create and manage Amazon EKS kubernetes cluster

## Description
Add new entry for eksctl.

## Motivation and Context

Fixes: #605

## How Has This Been Tested?

Run fine when trying it with the following command:

    arkade get eksctl

```
./bin/arkade get eksctl           
Downloading: eksctl
Downloading: https://github.com/weaveworks/eksctl/releases/download/v0.79.0/eksctl_Linux_amd64.tar.gz
18.64 MiB / 18.64 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
/tmp/eksctl_Linux_amd64.tar.gz written.
/tmp/eksctl eksctl
2022/01/19 12:06:22 extracted tarball into /tmp: 1 files, 0 dirs (487.370723ms)
2022/01/19 12:06:22 Extracted: /tmp/eksctl
2022/01/19 12:06:22 Copying /tmp/eksctl to /home/yannig/.arkade/bin/eksctl

Tool written to: /home/yannig/.arkade/bin/eksctl

# Add arkade binary directory to your PATH variable
export PATH=$PATH:$HOME/.arkade/bin/

# Test the binary:
/home/yannig/.arkade/bin/eksctl

# Or install with:
sudo mv /home/yannig/.arkade/bin/eksctl /usr/local/bin/
```

```
    file ~/.arkade/bin/eksctl
    /home/yannig/.arkade/bin/eksctl: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=4lrTx4ruPf-e2iY9i4an/iOTYPmA0intb1rIGOHs7/DBZ-vbBH8QxkEG0N2Yhh/xa1cm8B9u7EMjzBvIz-W, stripped
```

```
    type eksctl
    eksctl is /home/yannig/.arkade/bin/eksctl
```

```
    eksctl version
    0.79.0
```

MacOS + aarch64 download:

    ./bin/arkade get eksctl --os darwin --arch aarch64 

    Downloading: eksctl
    Downloading: https://github.com/weaveworks/eksctl/releases/download/v0.79.0/eksctl_Darwin_arm64.tar.gz
    20.27 MiB / 20.27 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
    /tmp/eksctl_Darwin_arm64.tar.gz written.
    /tmp/eksctl eksctl
    2022/01/19 12:12:00 extracted tarball into /tmp: 1 files, 0 dirs (578.130588ms)
    2022/01/19 12:12:00 Extracted: /tmp/eksctl
    2022/01/19 12:12:00 Copying /tmp/eksctl to /home/yannig/.arkade/bin/eksctl

    Tool written to: /home/yannig/.arkade/bin/eksctl

    # Add arkade binary directory to your PATH variable
    export PATH=$PATH:$HOME/.arkade/bin/

    # Test the binary:
    /home/yannig/.arkade/bin/eksctl

    # Or install with:
    sudo mv /home/yannig/.arkade/bin/eksctl /usr/local/bin/

```
    file ~/.arkade/bin/eksctl
    /home/yannig/.arkade/bin/eksctl: Mach-O 64-bit arm64 executable, flags:<|DYLDLINK|PIE>
```

MacOS download:

    ./bin/arkade get eksctl --os darwin

    Downloading: eksctl
    Downloading: https://github.com/weaveworks/eksctl/releases/download/v0.79.0/eksctl_Darwin_amd64.tar.gz
    19.76 MiB / 19.76 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
    /tmp/eksctl_Darwin_amd64.tar.gz written.
    /tmp/eksctl eksctl
    2022/01/19 12:12:08 extracted tarball into /tmp: 1 files, 0 dirs (553.76884ms)
    2022/01/19 12:12:08 Extracted: /tmp/eksctl
    2022/01/19 12:12:08 Copying /tmp/eksctl to /home/yannig/.arkade/bin/eksctl

    Tool written to: /home/yannig/.arkade/bin/eksctl

    # Add arkade binary directory to your PATH variable
    export PATH=$PATH:$HOME/.arkade/bin/

    # Test the binary:
    /home/yannig/.arkade/bin/eksctl

    # Or install with:
    sudo mv /home/yannig/.arkade/bin/eksctl /usr/local/bin/

```
    file ~/.arkade/bin/eksctl
    /home/yannig/.arkade/bin/eksctl: Mach-O 64-bit x86_64 executable
```

Windows download:

    ./bin/arkade get eksctl --os mingw

    Downloading: eksctl
    Downloading: https://github.com/weaveworks/eksctl/releases/download/v0.79.0/eksctl_Windows_amd64.zip
    18.91 MiB / 18.91 MiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
    /tmp/eksctl_Windows_amd64.zip written.
    Name: eksctl_Windows_amd64.zip, size: 19825514/tmp/eksctl.exe
    2022/01/19 12:12:13 extracted zip into /tmp: 1 files, 0 dirs (556.574877ms)
    2022/01/19 12:12:13 Extracted: /tmp/eksctl
    2022/01/19 12:12:13 Copying /tmp/eksctl to /home/yannig/.arkade/bin/eksctl

    Tool written to: /home/yannig/.arkade/bin/eksctl

    # Add arkade binary directory to your PATH variable
    export PATH=$PATH:$HOME/.arkade/bin/

    # Test the binary:
    /home/yannig/.arkade/bin/eksctl

    # Or install with:
    sudo mv /home/yannig/.arkade/bin/eksctl /usr/local/bin/

```
    file ~/.arkade/bin/eksctl.exe
    /home/yannig/.arkade/bin/eksctl.exe: PE32+ executable (console) x86-64 (stripped to external PDB), for MS Windows
```

## Are you a GitHub Sponsor (Yes/No?)

- [ ] Yes
- [X] No

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [X] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [X] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by Yannig** [comment]
Fix for this issue: https://github.com/alexellis/arkade/issues/605

**Review by alexellis — CHANGES_REQUESTED** [review]
Thanks for your submission. Generally, you will need to wait for approval before working on a PR.

But in this case, eksctl makes a lot of sense and we'd like to see this merged.

Before merge, we need you to add unit tests. See how this has been done in `get_test.go`

Here's an example that I wrote: https://github.com/alexellis/arkade/commit/fda8a74d6a8ee80f135cdd6474ed2fdcb1ed3bcc

Make sure it tests all the various download URLs available on the GitHub releases page and in your template.

Pay attention to darwin_arm64 for instance.

**Comment by Yannig** [comment]
Thanks @alexellis for your feedback.

The PR now have unit tests.

File: pkg/get/tools.go
Comment (alexellis) [inline pkg/get/tools.go:535]: Any reason for the different padding on this line?

**Comment by alexellis** [comment]
@Yannig under `How Has This Been Tested?` please show us that this works e2e with: `./arkade get eksctl`

You can also use --os and --arch to show us that the various URLs download a real file, followed by `file .arkade/bin/eksctl` for instance which prints out the file type.

File: pkg/get/get_test.go
Comment (alexellis) [inline pkg/get/get_test.go:930]: I know you think this is better, but these URLs are subject to change and don't always follow a pattern. Since we cannot control the project releases, please repeat each URL as per the example I sent you for KinD.

Using the full URL means we can copy/paste directly from a GitHub repo URL to compare. This indirection may been clever and more advanced, but in experience it works against us for maintenance.

File: pkg/get/tools.go
Comment (Yannig) [reply pkg/get/tools.go:535]: Sorry, It's an issue with the padding on my IDE. I'll fix this

File: pkg/get/get_test.go
Comment (Yannig) [reply pkg/get/get_test.go:930]: No problem. I will set the url directly.

**Comment by Yannig** [comment]
> @Yannig under `How Has This Been Tested?` please show us that this works e2e with: `./arkade get eksctl`
> 
> You can also use --os and --arch to show us that the various URLs download a real file, followed by `file .arkade/bin/eksctl` for instance which prints out the file type.

Done in the description of the PR.

**Comment by alexellis** [comment]
This looks great @Yannig - if CI passes I'll get it merged for you today.

Now that you know what we're looking for roughly, I expect your next contribution will be even quicker to get in.

**Comment by Yannig** [comment]
Thanks for the support @alexellis !

I'll follow this guidelines next time :)

**Pull request merged**
